tlatools/org.lamport.tlatools/customBuild.xml

```diff
@@ -146,14 +146,50 @@
 ==================================================================
 = This code uses sun.misc.Unsafe for manual memory management in =
 = the class src/tlc2/tool/fp/LongArray.java. This unsafe API is  =
-= marked as internal meaning it can change without warning. Java =
-= emits a lot of warnings about this and makes it essentially    =
-= impossible to silence them, so just ignore them. The unsafe    =
-= API is in the process of being standardized in newer versions  =
-= of Java and we will switch to that as soon as possible.        =
+= marked as internal meaning it can change without warning.      =
+= Ordinarily this would result in a lot of compiler warnings,    =
+= but to reduce noise these have been explicitly silenced. If    =
+= the build fails after upgrading Java, it might be because this =
+= API has been changed.                                          =
 ==================================================================
 		</echo>
 		<mkdir dir="${class.dir}" />
+		<!--
+			Separate compilation step for files which must have their warnings
+			uniformly ignored. Java 11 removed the -XDignore.symbol.file compiler
+			parameter that worked in Java 8, so this is the new method. For some
+			reason Java *really, really* does not want you to silence the warnings
+			around sun.misc.Unsafe; every method that works when suppressing every
+			other warning does not work here (SuppressWarning annotations,
+			-Xlint:none compiler arg, even -nowarn). So, we are reduced to
+			compiling the files we want to ignore with an actual command line call,
```

Comment:
I am not clear on why this has to be a command line call and not a `<javac>` task.

Reply:
There's no way to silence stderr output from a javac task.

Reply:
Ah. My fear is that `javac` comes from different places depending on the task used:
 - `<exec>` uses `javac` from `$PATH`
 - `<javac>` uses (as I understand it) the `javac` from the same JDK that Ant is being run with, unless you specify `build.compiler`, in which case that one is used.

That creates an awkward failure mode where setting `build.compiler` on the command line does not work as intended. Some of the sources will be compiled with a different compiler.

I know the warnings are verbose and mostly unnecessary, but perhaps this setup could be simpler if we just accept that they will be printed?

Reply:
Ah, fair enough. Okay I just switched it to a separate java precompile step. It still emits the sun.misc.Unsafe warnings but at least we can add `-Werror` to the main compilation step. This precompile step also includes a JavaCC-generated file to match your example.